src/ObjectUtils/ElfFile.cpp

```diff
@@ -54,16 +54,16 @@ class ElfFileImpl : public ElfFile {
   ElfFileImpl(std::filesystem::path file_path,
               llvm::object::OwningBinary<llvm::object::ObjectFile>&& owning_binary);
 
-  [[nodiscard]] ErrorMessageOr<ModuleSymbols> LoadSymbolsFromSymtab() override;
+  [[nodiscard]] ErrorMessageOr<ModuleSymbols> LoadSymbols() override;
   [[nodiscard]] ErrorMessageOr<ModuleSymbols> LoadSymbolsFromDynsym() override;
   [[nodiscard]] ErrorMessageOr<uint64_t> GetLoadBias() const override;
-  [[nodiscard]] bool HasSymtab() const override;
+  [[nodiscard]] bool HasSymbols() const override;
```

Comment:
same here.

Reply:
I don't get why `LoadSymbolsFromSymtab`, `HasSymtab` are renamed in the first place. The names are elf specific, but this is ElfFile.cpp so that is fine, right? Or do you intend to move them later into ObjectFile?

Reply:
Both are already in ObjectFile and are needed for core functionality that we need across both ELF and COFF. 

The idea is that LoadDebugSymbols is the method used by the UI to get symbols for a module if the symbols are embedded in the module (either as a symbol table or as DWARF). We currently use symtab for ELF files, but IMO this is already an implementation detail and we could just as well use DWARF if a symbol table is not present for whatever reason. 

My understanding is that LoadSymbolsFromDynsym is only used to inject code into the target process by finding the right functions in ld and/or libc (so we don't need to generalize this, yet). Correct?

Reply:
it is also used to normalize callstacks in the sampling report.

Reply:
We were also considering using Dynsym as a fallback when Symtab is not available.

Reply:
Yes, that would make sense to me. 

Side-note: I think the same can be done for PE/COFF (by looking at the export directory), but some care must be taken about forwarded symbols: A dll can forward a symbol to a different dll and not actually contain a definition for the symbol. It's not quite clear how we want to treat this in the UI, but a simple initial solution would be to just exclude forwarded symbols.

I'll take a note to look at adding this as a fallback, but will not make any additional changes in the current PR.

Reply:
@danielfenner I'll resolve this, but we can discuss if there are further concerns or ideas on how we should structure the code. I'll keep working on this part of the code.